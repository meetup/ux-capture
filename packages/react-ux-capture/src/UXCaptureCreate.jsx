// @flow
import React from 'react';

type callbackType = (label: string) => void;
type Props = {|
	onMark?: callbackType,
	onMeasure?: callbackType,
|};

// This component should be rendered in server markup - it will not work if only
// rendered client-side
export default ({ onMark, onMeasure }: Props) => {
	const uxCaptureConfigJS = `
		<script>
			if(window.UXCapture) {
				window.UXCapture.create({
					${onMark ? `"onMark": ${onMark.toString()},` : ''}
					${onMeasure ? `"onMeasure": ${onMeasure.toString()},` : ''}
				});
			}
		</script>
	`;

	return (
		<div dangerouslySetInnerHTML={{ __html: uxCaptureConfigJS }} /> // eslint-disable-line react/no-danger
	);
};
