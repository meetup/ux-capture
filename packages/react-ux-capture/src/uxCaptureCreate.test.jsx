import React from 'react';
import { shallow } from 'enzyme';
import UXCaptureCreate from './UXCaptureCreate';

const MOCK_PROPS = {
	onMark: foo => {
		if (foo) {
			return foo;
		}
	},
	onMeasure: bar => {
		if (bar) {
			return bar;
		}
	},
};

const renderComponent = (props = MOCK_PROPS) =>
	shallow(<UXCaptureCreate {...props} />);

describe('UXCaptureCreate', () => {
	it('renders correct markup when all props are defined', () => {
		expect(renderComponent()).toMatchSnapshot();
	});

	it('renders the correct markup when only onMeasure is provided', () => {
		const onMeasureProp = {
			onMeasure: MOCK_PROPS.onMeasure,
		};
		expect(renderComponent(onMeasureProp)).toMatchSnapshot();
	});

	it('renders the correct markup when only onMark is provided', () => {
		const onMarkProp = {
			onMark: MOCK_PROPS.onMark,
		};
		expect(renderComponent(onMarkProp)).toMatchSnapshot();
	});

	it('renders the correct markup when no props are provided', () => {
		expect(renderComponent({})).toMatchSnapshot();
	});

	it('renders the correct markup when unsupported props are provided', () => {
		const invalidProps = {
			...MOCK_PROPS,
			invalid: 'some prop',
		};
		expect(renderComponent(invalidProps)).toMatchSnapshot();
	});
});
