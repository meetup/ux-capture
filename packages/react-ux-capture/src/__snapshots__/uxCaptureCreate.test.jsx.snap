// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UXCaptureCreate renders correct markup when all props are defined 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UXCapture) {
				window.UXCapture.create({
					\\"onMark\\": function onMark(foo) {
    if (foo) {
      return foo;
    }
  },
					\\"onMeasure\\": function onMeasure(bar) {
    if (bar) {
      return bar;
    }
  },
				});
			}
		</script>
	",
    }
  }
/>
`;

exports[`UXCaptureCreate renders the correct markup when no props are provided 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UXCapture) {
				window.UXCapture.create({
					
					
				});
			}
		</script>
	",
    }
  }
/>
`;

exports[`UXCaptureCreate renders the correct markup when only onMark is provided 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UXCapture) {
				window.UXCapture.create({
					\\"onMark\\": function onMark(foo) {
    if (foo) {
      return foo;
    }
  },
					
				});
			}
		</script>
	",
    }
  }
/>
`;

exports[`UXCaptureCreate renders the correct markup when only onMeasure is provided 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UXCapture) {
				window.UXCapture.create({
					
					\\"onMeasure\\": function onMeasure(bar) {
    if (bar) {
      return bar;
    }
  },
				});
			}
		</script>
	",
    }
  }
/>
`;

exports[`UXCaptureCreate renders the correct markup when unsupported props are provided 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UXCapture) {
				window.UXCapture.create({
					\\"onMark\\": function onMark(foo) {
    if (foo) {
      return foo;
    }
  },
					\\"onMeasure\\": function onMeasure(bar) {
    if (bar) {
      return bar;
    }
  },
				});
			}
		</script>
	",
    }
  }
/>
`;
